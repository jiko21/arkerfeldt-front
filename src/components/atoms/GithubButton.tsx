/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { ButtonHTMLAttributes, FC } from 'react';
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const buttonStyle = css`
  background-color: #000000;
  border: none;
  border-radius: 6px;
  color: #ffffff;
  height: 30px;
  min-width: 240px;
  width: 100%;
  transition: 0.5s;
  &:hover {
    transform: translateY(-4px);
  }
`;

const innerButtonStyle = css`
  align-items: center;
  display: flex;
  font-size: 12px;
  font-weight: 700;
  justify-content: center;
  img {
    height: 24px;
    margin-right: 12px;
    width: 24px;
  }
`;

interface IProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  testId?: string;
}

const GithubButton: FC<IProps> = ({ testId, ...props }) => (
  <button css={buttonStyle} {...props} data-testid={testId}>
    <div css={innerButtonStyle}>
      <FontAwesomeIcon
        icon={faGithub}
        color="black"
        inverse
        css={css`
          margin-right: 8px;
        `}
      />
      Sign in with GitHub
    </div>
  </button>
);

export default GithubButton;
