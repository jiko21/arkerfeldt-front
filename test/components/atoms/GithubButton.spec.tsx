import GithubButton from '@/components/atoms/GithubButton';
import { fireEvent, render, screen } from '@testing-library/react';

describe('GithubButton.tsx', () => {
  let onClickMock: jest.Mock;
  beforeEach(() => {
    onClickMock = jest.fn();
  });

  afterEach(() => {
    onClickMock.mockClear();
  });
  it('should correctly render', () => {
    const { container } = render(
      <GithubButton onClick={onClickMock}>Sample</GithubButton>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should correctly call onClick', () => {
    const TEST_ID = 'TEST_ID';
    render(
      <GithubButton onClick={onClickMock} testId={TEST_ID}>
        Sample
      </GithubButton>,
    );
    fireEvent.click(screen.getByTestId(TEST_ID));
    expect(onClickMock).toBeCalled();
  });
});
