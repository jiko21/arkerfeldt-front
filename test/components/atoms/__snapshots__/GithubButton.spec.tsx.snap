// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubButton.tsx should correctly render 1`] = `
.emotion-0 {
  background-color: #000000;
  border: none;
  border-radius: 6px;
  color: #ffffff;
  height: 30px;
  min-width: 240px;
  width: 100%;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.emotion-0:hover {
  -webkit-transform: translateY(-4px);
  -moz-transform: translateY(-4px);
  -ms-transform: translateY(-4px);
  transform: translateY(-4px);
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
  font-weight: 700;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 img {
  height: 24px;
  margin-right: 12px;
  width: 24px;
}

.emotion-2 {
  margin-right: 8px;
}

<div>
  <button
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-github fa-w-16 fa-inverse emotion-2"
        color="black"
        data-icon="github"
        data-prefix="fab"
        focusable="false"
        role="img"
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          fill="currentColor"
        />
      </svg>
      Sign in with GitHub
    </div>
  </button>
</div>
`;
